scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt

```diff
@@ -383,9 +386,70 @@ internal class ScenarioRunServiceImpl(
     val workspace = workspaceService.findWorkspaceById(organizationId, workspaceId)
     sendScenarioRunMetaData(organizationId, workspace, scenarioId, scenarioRun.csmSimulationRun)
 
+    if (startInfo.runTemplate.deleteLastRunData == true) {
+      // Start a thread to check whether a scenariorun is completed
+      // then delete the previous simulation results
+      logger.debug("Start coroutine to poll simulation status")
+      GlobalScope.launch {
+        withTimeout(CSM_ARGO_WORKFLOWS_TIMEOUT.toLong()) {
+          deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(
+              scenarioRun, startInfo.scenario.lastRun)
+        }
+      }
+      logger.debug("Coroutine to poll simulation status launched")
+    }
     return scenarioRun.withoutSensitiveData()!!
   }
 
+  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(
+      currentRun: ScenarioRun,
+      lastRun: ScenarioLastRun?
+  ) {
+    while (true) {
+      val scenarioRunStatus = getScenarioRunStatus(currentRun.organizationId!!, currentRun.id!!)
+
+      if (scenarioRunStatus.state == ScenarioRunState.Successful) {
+        // deleteScenarioRunsByScenario delete all previous probes data, if it's what we want
```

Comment:
Yes this is what we want => no need to create and call the new method deleteLastRunSimulationData

Reply:
No more need need of lastRun also with deleteScenarioRunsByScenario